StudioTBA/CoronaIO
# Procedural generation of level

## What issue is this PR solving?
Closes: #3 
Closes: #45 

## Description of change
Given different prefabs, procedurally generate the map.

Things that could be improved include:
- Ability to set probabilities for building and obstruction prefabs

## How to reproduce the changes
1. Go to scene Game
2. Select Floor game object
3. Change the `Floor Scale` variable under `TileGenerator.cs` in the inspector to your liking. Must be greater than 20 and a multiple of 10
4. Start the scene
5. Click on the Floor game object again to see all the tiles that have been created

## Screenshots or GIFs
Different generations with floor scale of 30:

1. 
<img width="257" alt="Screen Shot 2020-03-27 at 8 34 08 PM" src="https://user-images.githubusercontent.com/10281272/77810356-75989680-706a-11ea-8974-33fb6dbe1fba.png">

2. 
<img width="256" alt="Screen Shot 2020-03-27 at 8 34 25 PM" src="https://user-images.githubusercontent.com/10281272/77810360-7a5d4a80-706a-11ea-81f3-4fc32b5effc5.png">

3. 
<img width="256" alt="Screen Shot 2020-03-27 at 8 34 39 PM" src="https://user-images.githubusercontent.com/10281272/77810368-7fba9500-706a-11ea-85d8-4f1b04080a53.png">
